src/gui/render.cpp

```diff
@@ -633,6 +634,43 @@ bool RENDER_UseSRGBFramebuffer()
 
 #endif
 
+#if C_OPENGL
+void log_warning_if_legacy_shader_name(const std::string name)
+{
+	static const std::map<std::string, std::string> legacy_name_mappings = {
+	        {"advinterp2x", "scaler/advinterp2x"},
+	        {"advinterp3x", "scaler/advinterp3x"},
+	        {"advmame2x", "scaler/advmame2x"},
+	        {"advmame3x", "scaler/advmame3x"},
+	        {"crt-easymode-flat", "crt/crt-easymode.tweaked"},
+	        {"crt-fakelottes-flat", "crt/fakelottes"},
+	        {"rgb2x", "scaler/rgb2x"},
+	        {"rgb3x", "scaler/rgb3x"},
+	        {"scan2x", "scaler/scan2x"},
+	        {"scan3x", "scaler/scan3x"},
+	        {"sharp", "interpolation/sharp"},
+	        {"tv2x", "scaler/tv2x"},
+	        {"tv3x", "scaler/tv3x"}};
+
+	std_fs::path path = name;
+	std_fs::path ext  = path.extension();
+
+	if (!(ext == "" || ext == ".glsl")) {
+		return;
+	}
```

Comment:
If someone uses `glshader = tv2x.glsl`, then we don't warn (`ext == ".glsl"`)?

Reply:
We do. We just exclude names here that do _not_ end with `.glsl` or are extensionless.

Then if the extension was `.glsl`, we remove it with `shader_path.replace_extension("");`.

This way the size of `legacy_name_mappings` could be cut in half (i.e. we don't need to list both names, once with and once without the `.glsl` extension). I just liked it better this way, but can include both versions in the map and remove this logic if you think it's too confusing.

Reply:
Thanks ; I misread the outer "not" applied to both `or` statements. Makes sense!